sed/calibrator/dld.py

```diff
@@ -0,0 +1,103 @@
+        sector_id_column (str, optional): Name of the column containing the
+            sectorID. Defaults to config["dataframe"]["sector_id_column"].
+        config (dict, optional): Configuration dictionary. Defaults to None.
+
+    Returns:
+        dask.dataframe.DataFrame: Dataframe with the new columns.
+    """
+    if tof_column is None:
+        if config is None:
+            raise ValueError("Either tof_column or config must be given.")
+        tof_column = config["dataframe"]["tof_column"]
+    if sector_id_column is None:
+        if config is None:
+            raise ValueError("Either sector_id_column or config must be given.")
+        sector_id_column = config["dataframe"]["sector_id_column"]
+
+    if sector_id_column in df.columns:
+        raise ValueError(
+            f"Column {sector_id_column} already in dataframe. " "This function is not idempotent.",
+        )
+    df[sector_id_column] = (df[tof_column] % 8).astype(np.int8)
+    df[tof_column] = (df[tof_column] // 8).astype(np.int32)
+    return df
+
+
+def align_dld_sectors(
+    df: dask.dataframe.DataFrame,
+    sector_delays: Sequence[float] = None,
+    sector_id_column: str = None,
+    tof_column: str = None,
+    config: dict = None,
+) -> Tuple[Union[pd.DataFrame, dask.dataframe.DataFrame], dict]:
+    """Aligns the 8s sectors to the first sector.
+
+    Args:
+        sector_delays (Sequece[float], optional): Sector delays for the 8s time.
+            in units of step. Calibration should be done with binning along dldTimeSteps.
+            Defaults to config["dataframe"]["sector_delays"].
+        sector_id_column (str, optional): Name of the column containing the
+            sectorID. Defaults to config["dataframe"]["sector_id_column"].
+        tof_column (str, optional): Name of the column containing the
+            time-of-flight. Defaults to config["dataframe"]["tof_column"].
+        config (dict, optional): Configuration dictionary. Defaults to None.
+
+    Returns:
+        dask.dataframe.DataFrame: Dataframe with the new columns.
+        dict: Metadata dictionary.
+    """
+    if sector_delays is None:
+        if config is None:
+            raise ValueError("Either sector_delays or config must be given.")
+        sector_delays = config["dataframe"]["sector_delays"]
+    if sector_id_column is None:
+        if config is None:
+            raise ValueError("Either sector_id_column or config must be given.")
+        sector_id_column = config["dataframe"]["sector_id_column"]
+    if tof_column is None:
+        if config is None:
+            raise ValueError("Either tof_column or config must be given.")
+        tof_column = config["dataframe"]["tof_column"]
+    # align the 8s sectors
+    sector_delays_arr = dask.array.from_array(sector_delays)
+
+    def align_sector(x):
+        val = x[tof_column] - sector_delays_arr[x[sector_id_column].values.astype(int)]
+        return val.astype(np.float32)
```

Comment:
no. This function serves only the purpouse here. it is not shifting an axis by values of an other. It is using the values of `sector_id_column` as index for the array `sector_delays_arr` which tells it how much it should shift. it very specific only for this application. The int is needed as it is used as an array index.

Reply:
What I mean is that you want some function to define/calibrate the values of shift of each sector. And this function conveniently could use the very same correction function for calculating the effect of test settings, if you allow the function to act also on binned data instead of only on dataframes. See e.g. the example for the energy correction, that's more or less how it is done there.

Reply:
The int is clear now, too many brackets...